src/reanimated2/animation/sequence.ts

```diff
@@ -1,29 +1,58 @@
-import { defineAnimation } from './util';
-import type { NextAnimation, SequenceAnimation } from './commonTypes';
+import { defineAnimation, getReduceMotionForAnimation } from './util';
+import type {
+  NextAnimation,
+  ReducedMotionConfig,
+  SequenceAnimation,
+} from './commonTypes';
 import type {
   Animation,
   AnimatableValue,
   AnimationObject,
   Timestamp,
 } from '../commonTypes';
 
-// TODO TYPESCRIPT This is a temporary type to get rid of .d.ts file.
-type withSequenceType = <T extends AnimatableValue>(...animations: T[]) => T;
+export function withSequence<T extends AnimatableValue>(
+  _reduceMotion: ReducedMotionConfig,
+  ...animations: T[]
+): T;
+
+export function withSequence<T extends AnimatableValue>(...animations: T[]): T;
 
-export const withSequence = function (
+export function withSequence(
+  _reduceMotion: ReducedMotionConfig | NextAnimation<AnimationObject>,
```

Comment:
1. Can we rename the variable so that it also mentions that this may be the first animation?
2. Does `withSequence()` (i.e. empty sequence) still work after these changes?
```suggestion
  _reduceMotionOrFirstAnimation: ReducedMotionConfig | NextAnimation<AnimationObject>,
```

Reply:
Empty sequence doesn't work, but it also didn't work before, since `animations[0]` is used. After some offline discussions I think it would be good to change it so the app doesn't crash when it's used, but also add some warning for the users.